File: BEE 1049.java
import java.io.IOException;
import java.util.Scanner;

public class Main{
    public static void main(String[] args) throws IOException{
        Scanner input = new Scanner(System.in);
        String a = input.nextLine();
        if(a.equals("vertebrado")){
            String b = input.nextLine();
            if(b.equals("ave")){
                String c = input.nextLine();
                if(c.equals("carnivoro"))
                    System.out.println("aguia");
                else if(c.equals("onivoro"))
                    System.out.println("pomba");
            }
            if(b.equals("mamifero")){
                String c = input.nextLine();
                if(c.equals("onivoro"))
                    System.out.println("homem");
                else if(c.equals("herbivoro"))
                    System.out.println("vaca");
            }
        }
        else if(a.equals("invertebrado")){
            String b = input.nextLine();
            if(b.equals("inseto")){
                String c = input.nextLine();
                if(c.equals("hematofago"))
                    System.out.println("pulga");
                else if(c.equals("herbivoro"))
                    System.out.println("lagarta");
            }
            if(b.equals("anelideo")){
                String c = input.nextLine();
                if(c.equals("hematofago"))
                    System.out.println("sanguessuga");
                else if(c.equals("onivoro"))
                    System.out.println("minhoca");
            }
        }
    }
}

File: BEE 1049.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    char a[22], b[22], c[22];
    cin>> a;
    if(strcmp(a, "vertebrado") == 0){
        cin>>b;
        if(strcmp(b, "ave") == 0){
            cin>>c;
            if(strcmp(c, "carnivoro") == 0)
                cout<<"aguia"<<endl;
            else if(strcmp(c, "onivoro") == 0)
                cout<<"pomba"<<endl;
        }
        if(strcmp(b, "mamifero") == 0){
            cin>>c;
            if(strcmp(c, "onivoro") == 0)
                cout<<"homem"<<endl;
            else if(strcmp(c, "herbivoro") == 0)
                cout<<"vaca"<<endl;
        }
    }
    else if(strcmp(a, "invertebrado") == 0){
        cin>>b;
        if(strcmp(b, "inseto") == 0){
            cin>>c;
            if(strcmp(c, "hematofago") == 0)
                cout<<"pulga"<<endl;
            else if(strcmp(c, "herbivoro") == 0)
                cout<<"lagarta"<<endl;
        }
        if(strcmp(b, "anelideo") == 0){
            cin>>c;
            if(strcmp(c, "hematofago") == 0)
                cout<<"sanguessuga"<<endl;
            else if(strcmp(c, "onivoro") == 0)
                cout<<"minhoca"<<endl;
        }
    }
    return 0;
}